src/control/Control.Layers.js

```diff
@@ -12,9 +12,14 @@ L.Control.Layers = L.Control.extend({
 	initialize: function (baseLayers, overlays, options) {
 		L.setOptions(this, options);
 
+		if (this.options.autoCollapse && !this.options.autoCollapseTimeout) {

```

Comment:
I think this could be simplified by just using one option `autoCollapse` which could accept a boolean (`true`/`false`) or a millisecond timeout. This would eliminate lines 15-18 entirely. I like the idea of a long timeout like 5 seconds by default.